multicluster/controllers/multicluster/core/leader_elector.go

```diff
@@ -73,20 +73,26 @@ func (m *remoteClusterManager) StartLeaderElection() (context.CancelFunc, error)
 				klog.Infof("Stopping leader election")
 				return
 			case <-ticker.C:
-				mutex.Lock()
+				m.mutex.Lock()
 				// 5 second timer has gone off, check if election must be done
 				if !m.needElection {
 					klog.Info("Check if leader election is necessary")
-					if m.electedLeaderCluster != nil && !m.electedLeaderCluster.IsConnected() {
-						klog.Infof("Leader disconnected", "leader", m.electedLeaderCluster.GetClusterID())
-						m.electedLeaderCluster = nil
+					if m.electedLeaderCluster != nil {
+						leaderIsConnected := m.electedLeaderCluster.IsConnected()
+						leaderId := m.electedLeaderCluster.GetClusterID()
+						klog.InfoS("Timer: ", "leader ", leaderId, "connected ", leaderIsConnected)
+						if !leaderIsConnected {
+							m.electedLeaderCluster = nil
+						}
 					}
 					if m.electedLeaderCluster == nil {
 						// do we have any member that is connected?
-						for _, cluster := range m.remoteClusters {
+						for id, cluster := range m.remoteClusters {
 							if cluster.IsConnected() {
-								klog.Info("Have atleast one connected leader")
+								klog.Info("leader is connected ", "cluster ", id)
```

Comment:
Ill remove these additional logs, I added them to debug tests, but its not needed.